python/test/gluon/test_frontend.py

```diff
@@ -657,7 +716,10 @@ def test_async_tma(fresh_knobs):
     shared_layout = ttgl.NVMMASharedLayout(swizzle_byte_width=128, element_bitwidth=16, rank=2)
     input_desc = TensorDescriptor.from_tensor(input, [XBLOCK, XBLOCK], shared_layout)
 
-    h = async_tma_kernel.warmup(input_desc, XBLOCK, grid=(1, ), num_warps=4)
+    target = get_cuda_test_aot_target(90)  # Hopper TMA
+    constexprs = build_constexpr_dict(XBLOCK=XBLOCK)
+    signature = build_signature_dict(input_desc=input_desc)
+    h = aot_compile_gluon(async_tma_kernel, target, constexprs, signature, num_warps=4, grid=(1, ))
```

Comment:
I like the idea to cross compile, but I'm not a fan of this interface. For CUDA we already have an `arch` compile flag which lets you override the SM version that we compile for. Perhaps you could add that for HIP as well and just pass `arch="gfx1200"`?

Reply:
Okay. I thought `TRITON_OVERRIDE_ARCH` is meant to override the architecture for the current backend. With that, it would mean we need to still anchor on the current GPU runtime to get the backend--tests won't be runnable on just bare CPUs as @ThomasRaoux would prefer and we need to filter different pytest cases per backend. I don't mind to do it if you think that's better. 

What you are suggesting in the above w `arch="hip:gfx1200"` actually would mean extending it to override the backend too.. That's a much larger change better to happen on its own actually.

Reply:
Looking further at how to potentially make `arch="hip:gfx1200"` work, it may need to intercept `driver.active` flow and force creating runtime not available on the current machine, which may run into weird issues given runtime creation can implicitly relies on machine states. Also IMO the `JITFunction.warmup/run()` kernel interface is designed for JIT and trying to make it support cross compilation w/ an env var changing it under the hood seems a bit strange. `triton.compile` is a more clear explicit interface for AOT compilation to me.

Reply:
@peterbell10, are you suggesting doing something like what is in here: https://github.com/triton-lang/triton/blob/phil/tma-repro/python/triton/_filecheck.py#L54
We could probably re-use this code and add support to provide a custom signature?

Reply:
My main concern is I don't want to have a re-implementation the JIT's specialization logic as it leaves the door open for there to be discrepancies between what we're testing and what actually happens in real code. 

If you strongly feel this should be a separate interface then a natural alternative would be to call the cross-compilation code from JITFunction, just with `driver.active.get_current_target()` as the target.